t/t.go

```diff
@@ -653,8 +667,81 @@ func downloadDataFiles() {
 	}
 }
 
+func createTestCoverageFile(path string) error {
+	outFile, err := os.Create(path)
+	if err != nil {
+		return err
+	}
+	defer outFile.Close()
+
+	cmd := command("echo", coverageFileHeader)
+	cmd.Stdout = outFile
+	if err := cmd.Run(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func fileExists(path string) bool {
+	fi, err := os.Stat(path)
+	if err != nil {
+		return false
+	}
+	return fi.Mode().IsRegular()
+}
+
+// Checks whether the test coverage file generated by go test is empty.
+// Empty coverage file are those that only has one line ("mode: <coverage_mode>") and nothing else.
+func isTestCoverageEmpty(path string) (bool, error) {
+	file, err := os.Open(path)
+	if err != nil {
+		return true, err
+	}
+	defer file.Close()
+
+	var l int
+	scanner := bufio.NewScanner(file)
+	for l = 0; scanner.Scan() && l <= 1; l++ {
+	}
+
+	if err = scanner.Err(); err != nil {
+		return true, err
+	}
+
+	return l <= 1, nil
+}
+
+func appendTestCoverageFile(src, des string) error {
+	if !fileExists(src) {
+		fmt.Println("src does not exist, skipping file")
+		return nil
+	}
+
+	isEmpty, err := isTestCoverageEmpty(src)
+	if err != nil {
+		return err
+	}
+	if isEmpty {
+		fmt.Println("no test files or no test coverage statement generated, skipping file")
+		return nil
+	}
+
+	cmd := command("bash", "-c", fmt.Sprintf("cat %s | grep -v \"%s\" >> %s", src, coverageFileHeader, des))
```

Comment:
@kevinmingtarja - we should change to this https://gobyexample.com/writing-files later on. I foresee a lot of issues.